src/org/ods/orchestration/util/MROPipelineUtil.groovy

```diff
@@ -324,6 +325,14 @@ class MROPipelineUtil extends PipelineUtil {
                         } else {
                             this.logger.debug("Repo '${repo.id}' is of type ODS Code Component. Nothing to do in phase '${name}' for target environment '${targetEnvToken}'.")
                         }
+                    } else if (repo.type?.toLowerCase() == PipelineConfig.REPO_TYPE_ODS_INFRA) {
+                        this.steps.stage('ODS Infrastructure Component') {
```

Comment:
The `stage` block should be removed. There are 6 stages: init, build, deploy, test, release, finalise. Having this `stage` block here would wrap the `ODS Infrastructure Component` inside the `deploy` stage (opened in `DeployStage`). I suspect this is a leftover from an old ODS version, which had substages for other repo types as well, which is now fixed.

Reply:
Resolved via aeec7628b3797aa33e7e369f884efe0c1a4c7203